spec.bs

```diff
@@ -1585,10 +1589,8 @@ the script or wasm, otherwise a [=tuple=] of ([=list=] of [=generated bids=],
     1. Let |timeDelta| be |auctionStartTime| minus |prevWin|'s [=previous win/time=].
     1. Set |timeDelta| to 0 if |timeDelta| is negative, |timeDelta|'s nearest second (rounding down)
       otherwise.
-    1. Let |prevWinIDL| be a new {{PreviousWin}}.
-    1. [=map/Set=] |prevWinIDL|["{{PreviousWin/timeDelta}}"] to |timeDelta|.
-    1. [=map/Set=] |prevWinIDL|["{{PreviousWin/adJSON}}"] to |prevWin|'s [=previous win/ad json=].
-    1. [=list/Append=] |prevWinIDL| to |prevWins|.
+    1. Let |prevWin| be the <code>[=sequence=]<{{PreviousWinElement}}></code> «|timeDelta|, |prevWin|'s [=previous win/ad=]».

```

Comment:
We cannot directly use [=previous win/ad=] as {{PreviousWinElement}}, since [[=previous win/ad=]](https://pr-preview.s3.amazonaws.com/brusshamilton/turtledove/pull/1258.html#previous-win) is an infra struct [[=interest group ad=]](https://pr-preview.s3.amazonaws.com/brusshamilton/turtledove/pull/1258.html#interest-group-ad), while {{PreviousWinElement}}'s ad is an IDL dictionary [{{AuctionAd}}](https://pr-preview.s3.amazonaws.com/brusshamilton/turtledove/pull/1258.html#dictdef-auctionad). See all the [conversions we had](https://github.com/WICG/turtledove/blob/main/spec.bs#L393) from {{AuctionAd}} to [=interest group ad=].

Do we store all fields of {{AuctionAd}} (including reporting ids) to previous wins ads field, without modifications needed? If so, we may add a new field to [[=interest group ad=]](https://pr-preview.s3.amazonaws.com/brusshamilton/turtledove/pull/1258.html#interest-group-ad) named something like "auction ad idl", and use that in previous wins. Similar to what we did for [auction config idl](https://wicg.github.io/turtledove/#auction-config-config-idl).

Reply:
We store the fields that we need to surface. Is it okay to leave some IDL fields undefined?

Reply:
yes, as long as it's not marked as "required".
But if we're not storing the entire AuctionAd in previous win, and want to explicitly make it clear about what fields are stored in previous wins, we can create a new IDL dictionary (e.g., PreviousWinAuctionAd), and we can create one PreviousWinAuctionAd from AuctionAd and store it into [=interest group=]'s ad.

Reply:
I think what we have is fine - we explicitly specify which fields we copy over when we create the previous win entry. I'm not sure what value having a separate IDL type would bring.